crates/relayer/relayer-components/src/chain/traits/queries/client_status.rs

```diff
@@ -0,0 +1,99 @@
+use cgp_core::prelude::*;
+
+use crate::chain::traits::types::client_state::{HasClientStateFields, HasClientStateType};
+use crate::chain::traits::types::consensus_state::{
+    HasConsensusStateFields, HasConsensusStateType,
+};
+use crate::chain::traits::types::ibc::HasIbcChainTypes;
+
+use super::chain_status::CanQueryChainStatus;
+use super::client_state::{CanQueryClientState, CanQueryClientStateWithHeight};
+use super::consensus_state::CanQueryConsensusState;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ClientStatus {
+    Frozen,
+    Expired,
+    Active,
+}
+
+#[async_trait]
+pub trait CanQueryClientStatus<Counterparty>:
+    HasIbcChainTypes<Counterparty>
+    + HasErrorType
+    + CanQueryClientState<Counterparty>
+    + CanQueryClientStateWithHeight<Counterparty>
+    + CanQueryChainStatus
+    + CanQueryConsensusState<Counterparty>
+where
+    Counterparty: HasIbcChainTypes<Self>
+        + HasClientStateType<Self>
+        + HasClientStateFields<Self>
+        + HasConsensusStateType<Self>
+        + HasConsensusStateFields<Self>,
+{
+    async fn query_client_status(
+        &self,
+        client_id: &Self::ClientId,
+    ) -> Result<ClientStatus, Self::Error>;
+}
+
+#[async_trait]
```

Comment:
FYI we no longer need to put `#[async_trait]` in `impl`s, as it is now natively supported in Rust. We only need it in `trait` definitions to desugar the `async fn` to return `impl Future + Send + Sync + 'static`.

Reply:
Let's clean those up all over the codebase in a follow-up PR.